src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorTableType.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public class ConnectorTableType : FormulaType
+    {
+        internal RecordType RecordType;
+
+        public ConnectorTableType(RecordType recordType)
+            : base(new ConnectorDType(recordType))
+        {
+            RecordType = recordType;
+        }
+
+        public override void Visit(ITypeVisitor vistor)
+        {
+            throw new System.NotImplementedException();
+        }
+    }
+
+    // Used in TabularIRVisitor
+    internal class ConnectorDType : DType
+    {
+        internal RecordType RecordType;      
+
+        internal ConnectorDType(RecordType recordType)
+            : base(DKind.Table, recordType._type.ToTable().TypeTree)
```

Comment:
>recordType._type.ToTable().TypeTree [](http://example.com/codeflow?start=32&length=35)

Key part that allows Texl functions to get the right return type (otherwise, we get ![] or *[])